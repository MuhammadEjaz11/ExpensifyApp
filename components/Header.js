import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter,Route, Routes, Link, NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { StartLogOut } from '../Actions/auth';

const Header = (props)=> (
    <div> 
        <h1>Header</h1> <br/>
        <NavLink to='/dashboard' activeclassname='active' >Go to Dashboard </NavLink> <br/>
        <NavLink to='/add'>Go to AddExpense </NavLink> <br/>
        <NavLink to='/edit'>Go to EditExpense </NavLink> <br/>
        <NavLink to='/help'>Go to Help </NavLink> <br/>
        <button onClick={props.StartLogOut}>Logout</button>
    </div>
);

const mapDispatchtoProps = (dispatch)=> ({
    StartLogOut: ()=> (dispatch(StartLogOut()))
})

export default connect(undefined, mapDispatchtoProps)(Header);
