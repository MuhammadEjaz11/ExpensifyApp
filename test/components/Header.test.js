import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';

test('to cehck the Header Componen',()=>{
    const renderer = new ShallowRenderer();
    renderer.render(<Header/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})