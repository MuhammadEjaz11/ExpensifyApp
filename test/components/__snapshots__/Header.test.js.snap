// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to cehck the Header Componen 1`] = `
<div>
  <h1>
    Header
  </h1>
   
  <br />
  <NavLink
    activeclassname="active"
    to="/"
  >
    Go to Dashboard 
  </NavLink>
   
  <br />
  <NavLink
    to="/add"
  >
    Go to AddExpense 
  </NavLink>
   
  <br />
  <NavLink
    to="/edit"
  >
    Go to EditExpense 
  </NavLink>
   
  <br />
  <NavLink
    to="/help"
  >
    Go to Help 
  </NavLink>
   
  <br />
</div>
`;
